crates/interledger-packet/src/oer.rs

```diff
@@ -170,55 +172,76 @@ impl<'a> BufOerExt<'a> for &'a [u8] {
         }
     }
 
-    fn read_variable_length_timestamp<L>(&mut self) -> Result<VariableLengthTimestamp<L>> {
+    fn read_variable_length_timestamp<L: SmallVariableLengthField>(
+        &mut self,
+    ) -> Result<VariableLengthTimestamp<L>> {
         let regex = regex::bytes::Regex::new(r"^[0-9]{4}[0-9]{2}{5}(\.[0-9]{1,3})$").unwrap();
```

Comment:
```suggestion
        let regex = regex::bytes::Regex::new(r"^[0-9]{4}[0-9]{2}{5}(\.[0-9]{1,3})?$").unwrap();
```

I think the fractional were optional? Or at least they are handled as such in the "trimming phase". Yeah, they are: https://github.com/interledger/rfcs/blob/master/0030-notes-on-oer-encoding/0030-notes-on-oer-encoding.md#decoding-1